dev/com.ibm.ws.jaxrs.2.x.monitor/src/com/ibm/ws/jaxrs/monitor/JaxRsMonitorFilter.java

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jaxrs.monitor;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.servlet.GenericServlet;
+import javax.servlet.ServletContext;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.ext.Provider;
+
+import com.ibm.websphere.csi.J2EEName;
+import com.ibm.websphere.monitor.annotation.Monitor;
+import com.ibm.websphere.monitor.annotation.PublishedMetric;
+import com.ibm.websphere.monitor.meters.MeterCollection;
+import com.ibm.ws.runtime.metadata.ComponentMetaData;
+import com.ibm.ws.runtime.metadata.ModuleMetaData;
+import com.ibm.ws.threadContext.ComponentMetaDataAccessorImpl;
+
+
+/**
+ * Monitor Class for RESTful Resource Methods.
+ */ 
+@Monitor(group = "RESTful")
+@Provider
+public class JaxRsMonitorFilter implements ContainerRequestFilter, ContainerResponseFilter {
+
+    @Context
+    ResourceInfo resourceInfo;
+    
+    // jaxRSCountByName is a MeterCollection that will hold the RESTfulStats MXBean for each RESTful
+    // resoure method
+    @PublishedMetric
+    public MeterCollection<RESTful_Stats> jaxRsCountByName = new MeterCollection<RESTful_Stats>("RESTful",this);
+    
+    // appMetricInfos is a hashmap used to store information for runtime and cleanup at 
+    // application stop time.
+    protected ConcurrentHashMap<String,RestfulMetricInfo> appMetricInfos = new ConcurrentHashMap<String,RestfulMetricInfo>();
```

Comment:
```suggestion
    ConcurrentHashMap<String,RestfulMetricInfo> appMetricInfos = new ConcurrentHashMap<String,RestfulMetricInfo>();
```

I think package visibility is sufficient here.  No need for it to be `protected`.

Reply:
Agreed.